worksheet_D/oxo.py

```diff
@@ -0,0 +1,176 @@
+BOARD_EDGE_LENGTH = 3
+WINNING_ROW_LENGTH = 3
+
+# Changing the board size obviously requires redefining the --+---+-- drawing section in show().
+# The winning row may not have to be the BOARD_EDGE_LENGTH so this is built in to the checks.
+#
+# oxotest.py does not allow lengths greater than 3 currently. I may tweak it to prove it works.
+#
+# Also would have written as BOARD_HEIGHT and BOARD_WIDTH but the worksheet asks for n * n implying square.
```

Comment:
Thanks for pointing this out, I'll change it to m * n for next year's students :)